Tickets/dm 38670

pcortes80 opened the pull request:
Create `ServerInstancesTest` directory with test cases in order to understand why TCP/IP module can not run multiple instances.

teweitsai reviewed (changes requested):
Let us find a time to discuss the followings:

[Overview]

1. For the new vis in this PR, the font size of icon should be 9 instead of 8 or others. After modifying the font size to be consistent with others, need to check the icon text to be clear or not.

2. I think the tests should have:
(a) Two servers can construct the connections at the same time.
(b) Two servers can have the multiple times of connections at the same time.
(c) User can stop two servers without the problem at the same time.
(d) Two clients can write commands to two servers at the same time.
(e) Two clients can write events to two servers at the same time.
(f) Two clients can write telemetry to two servers at the same time.
(g) Two servers can write command status to two clients at the same time.
(h) Two servers can write event to two clients at the same time.
(i) Two servers can write telemetry to two clients at the same time.
(j) Two servers can report the connection status at the same time.
(k) Two servers can report the log messages at the same time.


[startTcpServerA.vi]

1. This vi can call the setDefaultServerA.vi. Right?


[startTcpServerB.vi]

1. This vi can call the setDefaultServerB.vi. Right?


[testAll.vi]

1. Let us change the order to test all the functions of single instance first followed by the intergration test and multiple instances test.


[testConfigServerAandB.vi]

1. You can reuse the setDefaultServerA.vi and setDefaultServerB.vi. Right?

2. Maybe reorganize the implementation in the block diagram to be clean and avoid the overlap.


[testCreateTCPIPConnectionServersAandB.vi]

1. Maybe rename to testCreateTcpIpConnectionServersAandB.vi.

2. The calling of "Define Test.vi" is weird???

3. Why there is no any assertion vi to verify the connection is constructed or closed? In addition, the "2_stop" case in the while loop can not really verify the connection is closed or not after the construction. For the checking of connection status, maybe from the client's and server's viewpoints together.


[testGetConnStatusFromServersAandB.vi]

1. Duplication from testCreateTCPIPConnectionServersAandB.vi? Why not combine them into one vi?

2. I do not understanad the labels: T, F.

3. Please explain the assertion data. I do not understand at all.

4. I think you need to assert the condition that both servers A and B are connected. Do you have it? It looks like the test case is A and B will have the connection individually but not the same time. The waiting time is too short in clients to let me have this suspicion.


[testRegEvtServersAandB.vi]

1. The implementation is for the command actually.


[testSendCmdStatusServersAandB.vi]

1. The text icon (Sd Cm) is misleading unless people open the icon setting... This applies to other vis. Actually, you need to use the appropriate abbreviation. For example, "Sv" is not correct for "Server":
https://www.abbreviations.com/abbreviation/server

2. You should destroy two user event instances instead of one.


[testSendEvtStatusServersAandB.vi]

1. You should destroy two user event instances instead of one.

2. Don't you feel the empty new line in event string is weird?


[testTriggerGenUserEventServerAandB.vi]

1. Do you need the boolean array indicators in the front panel?

2. I do not understand this vi at all. Why does the TcpIp server need to get the event of connection status???
 

[serverInstancesTest.vi]

1. The failed tests should run in the final.

pcortes80 commented:
Let us find a time to discuss the followings:

[Overview]

1. For the new vis in this PR, the font size of icon should be 9 instead of 8 or others. After modifying the font size to be consistent with others, need to check the icon text to be clear or not.
**Reply: Done**

2. I think the tests should have:
(a) Two servers can construct the connections at the same time. **DONE**
(b) Two servers can have multiple times of connections at the same time.
(c) User can stop two servers without the problem at the same time.
(d) Two clients can write commands to two servers at the same time.
(e) Two clients can write events to two servers at the same time.
(f) Two clients can write telemetry to two servers at the same time.
(g) Two servers can write command status to two clients at the same time.
(h) Two servers can write events to two clients at the same time.
(i) Two servers can write telemetry to two clients at the same time.
(j) Two servers can report the connection status at the same time.
(k) Two servers can report the log messages at the same time.

[startTcpServerA.vi]

1. This vi can call the setDefaultServerA.vi. Right?
**Reply: Done**

[startTcpServerB.vi]

1. This vi can call the setDefaultServerB.vi. Right?
**Reply: Done**

[testAll.vi]

1. Let us change the order to test all the functions of single instance first followed by the intergration test and multiple instances test.
**Reply: This will be done in the Phase 2 of ticket DM-38670.**

[testConfigServerAandB.vi]

1. You can reuse the setDefaultServerA.vi and setDefaultServerB.vi. Right?
Maybe reorganize the implementation in the block diagram to be clean and avoid the overlap.
**Reply: Done**

[testCreateTCPIPConnectionServersAandB.vi]

1. Maybe rename to testCreateTcpIpConnectionServersAandB.vi.
**Reply: Done**

2. The calling of "Define Test.vi" is weird???
**Reply: Checked in the last call.**

3. Why there is no any assertion vi to verify the connection is constructed or closed? In addition, the "2_stop" case in the while loop can not really verify the connection is closed or not after the construction. For the checking of connection status, maybe from the client's and server's viewpoints together.
**Reply: I modified this test to focus on the client’s view point. The clients will try to construct the TCP/IP connection after both TCP server run and then the test will check whether the connection was constructed or not. With this modification, the while loop was not necessary.** 

[testGetConnStatusFromServersAandB.vi]

1. Duplication from testCreateTCPIPConnectionServersAandB.vi? Why not combine them into one vi?

2. I do not understand the labels: T, F.

3. Please explain the assertion data. I do not understand at all.

4. I think you need to assert the condition that both servers A and B are connected. Do you have it? It looks like the test case is A and B will have the connection individually but not the same time. The waiting time is too short in clients to let me have this suspicion.
 
**Reply: I preferred to remove this test because I duplicated from testCreateTCPIPConnectionServersAandB.vi**

[testRegEvtServersAandB.vi]

1. The implementation is for the command actually. 
**Reply: I modified this vi replacing the regCmd.vi with regEvt.vi.**

[testSendCmdStatusServersAandB.vi]

1. The text icon (Sd Cm) is misleading unless people open the icon setting... This applies to other vis. Actually, you need to use the appropriate abbreviation. For example, "Sv" is not correct for "Server": https://www.abbreviations.com/abbreviation/server
**Reply: Done.**

2. You should destroy two user event instances instead of one.
**Reply: Done.**

[testSendEvtServersAandB.vi]

1. You should destroy two user event instances instead of one.
**Reply: Done.**

2. Don't you feel the empty new line in event string is weird?
**Reply: Fixed.**

[testTriggerGenUserEventServerAandB.vi]

1. Do you need the boolean array indicators in the front panel?
 
2. I do not understand this vi at all. Why does the TcpIp server need to get the event of connection status???
 
**Reply: I decided to remove this vi.**

[serverInstancesTest.vi]

1. The failed tests should run in the final.
**Reply: Fixed.**
